### PR TITLE
ddl: decouple executor part out from `ddl`

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #54436

Problem Summary:

### What changed and how does it work?
- decouple executor part out from `ddl`, as this PR is quite large, some part is still coupled, will do it in next pr
- this pr is large mostly due to changes of rename such as from `(d *ddl)` to `(e *executor)`, and callsite of domain.DDL -> domain.DDLExecutor

### Check List

Tests 

- [x] Unit test, this PR only fix existing tests, no need to add more as this PR only re-structure existing code
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/executor_test.go [hunk 1]
```diff
@@ -15,26 +15,319 @@
...
 	"github.com/pingcap/tidb/pkg/store/mockstore"
 	"github.com/pingcap/tidb/pkg/testkit"
+	"github.com/pingcap/tidb/pkg/testkit/testfailpoint"
 	"github.com/pingcap/tidb/pkg/util"
+	"github.com/pingcap/tidb/pkg/util/chunk"
 	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
 )
 
+func TestGetDDLJobs(t *testing.T) {
```
Comment (D3Hunter) [inline pkg/ddl/executor_test.go:47]: moved from prev ddl_api_test.go

File: pkg/ddl/ddl_api_test.go
Comment (D3Hunter) [inline pkg/ddl/ddl_api_test.go:1]: move to executor_test.go

File: pkg/ddl/executor.go [hunk 2]
```diff
@@ -9627,8 +9743,21 @@ func (d *ddl) AlterCheckConstraint(ctx sessionctx.Context, ti ast.Ident, constrN
...
 
-	err = d.DoDDLJob(ctx, job)
-	err = d.callHookOnChanged(job, err)
+	err = e.DoDDLJob(ctx, job)
+	err = e.callHookOnChanged(job, err)
+	return errors.Trace(err)
+}
+
+// TODO remove it later.
+func (e *executor) callHookOnChanged(job *model.Job, err error) error {
```
Comment (D3Hunter) [inline pkg/ddl/executor.go:9752]: duplicate with what's in `ddl`, will change it later, this pr is very large now.

**Review by lance6716 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Pull request merged**